File: IPL_TICKET_BOOKING_SYSTEM_INTADM21DN003_POD1_AES_Project_14-master/IPL_TBS.BusinessLogic/MatchDetailsService.cs
using IPL_TBS.DataAccess.Entities;
using System;
using System.Collections.Generic;
using System.Data.Common;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using IPL_TBS.DataAccess;

namespace IPL_TBS.BusinessLogic
{
	public class MatchDetailsService : IDisposable
	{
		private readonly RegisterDBContext context;
		public MatchDetailsService()
		{
			context = new RegisterDBContext();
		}
		public void Dispose()
		{
			context.Dispose();
		}

		public void Post(string Location, string Date, string Battle)
		{
			try
			{
				MatchDetails obj1 = new MatchDetails();
				obj1.Location = Location;
				obj1.Date = Date;
				obj1.Battle = Battle;
				context.MatchDetail.Add(obj1);
			}
			catch (DbException ex)
			{
				throw new IPLException("Error Reading Data", ex);

			}
			catch (Exception ex)
			{
				throw new IPLException("Unknown Error while reading Stadiumdetails", ex);

			}
		}
		public List<MatchDetails> GetAllActive()
		{
			try
			{
				var items = from item in context.MatchDetail
								//where item.IsActive
							select item;
				return items.ToList();
			}
			catch (DbException ex)
			{
				throw new IPLException("Error reading data", ex);
			}
			catch (Exception ex)
			{
				throw new IPLException("Unknown error while reading Match Details", ex);
			}
		}
	}
}

File: IPL_TICKET_BOOKING_SYSTEM_INTADM21DN003_POD1_AES_Project_14-master/IPLAngular/src/app/shared/match-details.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
@Injectable({
  providedIn: 'root'
})
export class MatchDetailsService {
  readonly url="http://localhost:51985/Api/MatchDetails/Add_MatchDetails";
  constructor(private http:HttpClient) {

   }
   addmatch(value:any){
    const details={
    //Match_Id:value.Match_Id,
      Location:value.Location,
      Date:value.Date,
      Battle:value.Battle
     
    }
    console.log(details);
    //return data;
   return this.http.post(this.url,details);
  }
}
